### DISCUSSION TITLE
[Idea] add to the carousels the possibility of showing only products available for orders

### DISCUSSION THREAD

**Discussion by Maofree** [PrestaShop/PrestaShop#35531] [Ideas]
### The problem

Hi
some companies that I follow have pointed out to me that the carousels on the homepage should only show products that have availability, it actually makes no sense to advertise products that cannot currently be purchased. It would be necessary to add an option, or directly put an option in the sql call, to filter only products with quantities if warehouse management is active.

a similar option has recently been added to the ps_facedsearch module, where it moves those without quantities to the queue, and I find it makes sense

the homepage carousel modules would be:
ps_specials
ps_newproducts
ps_bestsellers
ps_featuredproducts

product carousel modules:
ps_categoryproducts
ps_crossselling

I think it's useful for everyone
the problem is very serious, with very large sites, connected to management systems, where the quantities vary quickly, and the carousels should only show products that can be purchased automatically

bye

### The solution

for these modules
ps_specials
ps_newproducts
ps_bestsellers
ps_featuredproducts
ps_categoryproducts
ps_crossselling

Adding an option to give the possibility to show only products with quantity if warehouse management is enabled

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by MatShir** [comment]
Sounds good, the option would be placed on each module I guess ?

**Comment by Maofree** [comment]
Hi
yes

**Reply by MatShir** [reply]
👌

**Comment by Maofree** [comment]
Hi @MatShir 
Je dois régler d'autres choses, la semaine prochaine j'essaierai d'ajouter cette option, j'ai besoin de deux informations :
- pensez-vous qu'il est préférable d'avoir la possibilité de supprimer les produits indisponibles ou de les mettre en file d'attente comme le fait FaceSearch ? Je préférerais les supprimer étant un carrousel
- pour apporter des modifications aux 6 modules, dois-je faire 6 PR différents ou juste 1 avec les 6 modifications ? Je le ferais certainement sur un seul module, puis lorsque les modifications seraient acceptées, je les ferais également sur les autres.

thanks

**Reply by MatShir** [reply]
No worries, 

> pensez-vous qu'il est préférable d'avoir la possibilité de supprimer les produits indisponibles ou de les mettre en file d'attente comme le fait FaceSearch ? Je préférerais les supprimer étant un carrousel

I'm good with removing them for now, maybe we will improve the feature overtime. It won't impact users since they would need to enable the feature.

> pour apporter des modifications aux 6 modules, dois-je faire 6 PR différents ou juste 1 avec les 6 modifications ? Je le ferais certainement sur un seul module, puis lorsque les modifications seraient acceptées, je les ferais également sur les autres.

You will have to do a PR per module. Let's start with one, good idea ;)
